Sources/NIOCore/ByteBuffer-hexdump.swift

```diff
@@ -0,0 +1,188 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+extension ByteBuffer {
+
+    /// Describes a ByteBuffer hexDump format.
+    /// Can be either xxd output compatible, or hexdump compatible.
+    public struct HexDumpFormat {
+        enum Value {
```

Comment:
```suggestion
        enum Value: Sendable {
```

Reply:
Went with @Lukasa's feedback and made them both `Hashable` and `Sendable`.